libamqpprox/amqpprox_limitcontrolcommand.cpp

```diff
@@ -246,68 +247,115 @@ void handleDataRateLimit(
     ControlCommandOutput<ControlCommand::OutputFunctor> &output)
 {
-    auto alarmLimiter =
+    bool anyConfiguredLimit = false;
+
+    auto alarmConnRateLimiter =
         connectionLimiterManager->getAlarmOnlyConnectionRateLimiter(vhostName);
-    if (alarmLimiter) {
+    if (alarmConnRateLimiter) {
         output << "Alarm only limit, for vhost " << vhostName << ", "
-               << alarmLimiter->toString() << ".\n";
+               << alarmConnRateLimiter->toString() << ".\n";
+        anyConfiguredLimit = true;
     }
-
-    auto limiter =
+    else {
+        std::optional<uint32_t> alarmConnRateLimit =
+            connectionLimiterManager->getAlarmOnlyDefaultConnectionRateLimit();
+        if (alarmConnRateLimit) {
+            output << "Alarm only limit, for vhost " << vhostName
+                   << ", allow average " << *alarmConnRateLimit
+                   << " number of connections per second.\n";
+            anyConfiguredLimit = true;
+        }
+    }
+    auto connRateLimiter =
         connectionLimiterManager->getConnectionRateLimiter(vhostName);
-    if (limiter) {
-        output << "For vhost " << vhostName << ", " << limiter->toString()
-               << ".\n";
+    if (connRateLimiter) {
+        output << "For vhost " << vhostName << ", "
+               << connRateLimiter->toString() << ".\n";
+        anyConfiguredLimit = true;
     }
-
-    if (!alarmLimiter && !limiter) {
-        std::optional<uint32_t> alarmOnlyConnectionRateLimit =
-            connectionLimiterManager->getAlarmOnlyDefaultConnectionRateLimit();
-        std::optional<uint32_t> connectionRateLimit =
+    else {
+        std::optional<uint32_t> connRateLimit =
             connectionLimiterManager->getDefaultConnectionRateLimit();
-        if (alarmOnlyConnectionRateLimit || connectionRateLimit) {
-            if (alarmOnlyConnectionRateLimit) {
-                output << "Alarm only limit, for vhost " << vhostName
-                       << ", allow average " << *alarmOnlyConnectionRateLimit
-                       << " number of connections per second.\n";
-            }
-            if (connectionRateLimit) {
-                output << "For vhost " << vhostName << ", allow average "
-                       << *connectionRateLimit
-                       << " number of connections per second.\n";
-            }
-        }
-        else {
-            output << "No default connection rate limit configured "
-                      "for any vhost.\n";
+        if (connRateLimit) {
+            output << "For vhost " << vhostName << ", allow average "
+                   << *connRateLimit << " number of connections per second.\n";
+            anyConfiguredLimit = true;
         }
     }
+
+    std::size_t alarmDataRateLimit =
+        dataRateLimitManager->getDataRateAlarm(vhostName);
+    if (alarmDataRateLimit == std::numeric_limits<std::size_t>::max()) {
+        alarmDataRateLimit = dataRateLimitManager->getDefaultDataRateAlarm();
```

Comment:
if we enter this block, there is no default data rate alarm set. `getDataRateAlarm` / `getDataRateLimit` do this default fallback already